sqlglot/schema.py

```diff
@@ -49,37 +60,49 @@ class MappingSchema(Schema):
         dialect (str): The dialect to be used for custom type mappings.
     """
 
-    def __init__(self, schema, visible=None, dialect=None):
-        self.schema = schema
+    def __init__(self, schema=None, visible=None, dialect=None):
+        self.schema = schema or {}
         self.visible = visible
         self.dialect = dialect
         self._type_mapping_cache = {}
-
-        depth = _dict_depth(schema)
-
-        if not depth:  # {}
-            self.supported_table_args = []
-        elif depth == 2:  # {table: {col: type}}
-            self.supported_table_args = ("this",)
-        elif depth == 3:  # {db: {table: {col: type}}}
-            self.supported_table_args = ("db", "this")
-        elif depth == 4:  # {catalog: {db: {table: {col: type}}}}
-            self.supported_table_args = ("catalog", "db", "this")
-        else:
-            raise OptimizeError(f"Invalid schema shape. Depth: {depth}")
-
-        self.forbidden_args = {"catalog", "db", "this"} - set(self.supported_table_args)
+        self.supported_table_args = []
+        self.forbidden_args = []
```

Comment:
Shall we call this `forbidden_table_args` to clarify and to be consistent with `supported_table_args`?